comment-spacings false positives for nolint directives

**Describe the bug**
I'm using revive with golangci-lint and since the latest update I'm getting errors like this:
```
api/common/validate/validate.go:15:1: comment-spacings: no space between comment delimiter and comment text (revive)
//nolint:gochecknoglobals
```
I've updated the config to contain `nolint` in the `comment-spacings` arguments to make it go away but it's odd as this was not happening before.

**To Reproduce**
Steps to reproduce the behavior:
1. I updated golangci-lint:
```sh
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
```

2. I run it with the following flags & configuration file:
```shell
golangci-lint run
```

```yaml
linters:
  disable-all: true
  enable:
    - revive
linters-settings:
  revive:
    rules:
      - name: comment-spacings
        disabled: false
```


**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.2 LTS
 - go version go1.20.2 linux/amd64

--- a/rule/comment-spacings.go
+++ b/rule/comment-spacings.go
@@ -22,7 +22,7 @@ func (r *CommentSpacingsRule) configure(arguments lint.Arguments) {
 	defer r.Unlock()
 
 	if r.allowRe == nil {
-		r.allowRe = regexp.MustCompile("//(line|extern|export|[a-z0-9]+:[a-z0-9])") // see https://go-review.googlesource.com/c/website/+/442516/1..2/_content/doc/comment.md#494
+		r.allowRe = regexp.MustCompile("//(line |extern |export |[a-z0-9]+:[a-z0-9])") // see https://go-review.googlesource.com/c/website/+/442516/1..2/_content/doc/comment.md#494
 		r.allowList = []string{}
 		for _, arg := range arguments {
 			allow, ok := arg.(string) // Alt. non panicking version

--- a/rule/comment-spacings.go
+++ b/rule/comment-spacings.go
@@ -2,6 +2,7 @@ package rule
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 	"sync"
 
@@ -12,6 +13,7 @@ import (
 // the comment symbol( // ) and the start of the comment text
 type CommentSpacingsRule struct {
 	allowList []string
+	allowRe   *regexp.Regexp
 	sync.Mutex
 }
 
@@ -19,13 +21,9 @@ func (r *CommentSpacingsRule) configure(arguments lint.Arguments) {
 	r.Lock()
 	defer r.Unlock()
 
-	if r.allowList == nil {
-		r.allowList = []string{
-			"//go:",
-			"//revive:",
-			"//nolint:",
-		}
-
+	if r.allowRe == nil {
+		r.allowRe = regexp.MustCompile("//(line|extern|export|[a-z0-9]+:[a-z0-9])") // see https://go-review.googlesource.com/c/website/+/442516/1..2/_content/doc/comment.md#494
+		r.allowList = []string{}
 		for _, arg := range arguments {
 			allow, ok := arg.(string) // Alt. non panicking version
 			if !ok {
@@ -87,5 +85,5 @@ func (r *CommentSpacingsRule) isAllowed(line string) bool {
 		}
 	}
 
-	return false
+	return r.allowRe.MatchString(line)
 }

--- a/rule/comment-spacings.go
+++ b/rule/comment-spacings.go
@@ -2,7 +2,6 @@ package rule
 
 import (
 	"fmt"
-	"regexp"
 	"strings"
 	"sync"
 
@@ -13,7 +12,6 @@ import (
 // the comment symbol( // ) and the start of the comment text
 type CommentSpacingsRule struct {
 	allowList []string
-	allowRe   *regexp.Regexp
 	sync.Mutex
 }
 
@@ -21,8 +19,7 @@ func (r *CommentSpacingsRule) configure(arguments lint.Arguments) {
 	r.Lock()
 	defer r.Unlock()
 
-	if r.allowRe == nil {
-		r.allowRe = regexp.MustCompile("//(line |extern |export |[a-z0-9]+:[a-z0-9])") // see https://go-review.googlesource.com/c/website/+/442516/1..2/_content/doc/comment.md#494
+	if r.allowList == nil {
 		r.allowList = []string{}
 		for _, arg := range arguments {
 			allow, ok := arg.(string) // Alt. non panicking version
@@ -85,5 +82,5 @@ func (r *CommentSpacingsRule) isAllowed(line string) bool {
 		}
 	}
 
-	return r.allowRe.MatchString(line)
+	return isDirectiveComment(line)
 }

--- a/rule/utils.go
+++ b/rule/utils.go
@@ -165,3 +165,9 @@ func checkNumberOfArguments(expected int, args lint.Arguments, ruleName string) 
 		panic(fmt.Sprintf("not enough arguments for %s rule, expected %d, got %d. Please check the rule's documentation", ruleName, expected, len(args)))
 	}
 }
+
+var directiveCommentRE = regexp.MustCompile("//(line |extern |export |[a-z0-9]+:[a-z0-9])") // see https://go-review.googlesource.com/c/website/+/442516/1..2/_content/doc/comment.md#494
+
+func isDirectiveComment(line string) bool {
+	return directiveCommentRE.MatchString(line)
+}

--- a/testdata/comment-spacings.go
+++ b/testdata/comment-spacings.go
@@ -48,8 +48,7 @@ type c struct {
 	d *int `json:"d,omitempty"`
 }
 
-//line
-//extern
-//export
+//extern open
+//export MyFunction
 
 //nolint:gochecknoglobals

--- a/testdata/comment-spacings.go
+++ b/testdata/comment-spacings.go
@@ -47,3 +47,9 @@ type c struct {
 	//+optional
 	d *int `json:"d,omitempty"`
 }
+
+//line
+//extern
+//export
+
+//nolint:gochecknoglobals

